### PR TITLE
feat(rome_js_formatter): Refine variable declaration formatting

### PR DESCRIPTION
**Pull request by MichaReiser**
…s formatting

This PR ensures that our formatting closer matche's prettiers variable declaration formatting by:

* `Indent`ing the first declarator if there are multiple declarators, or it has a leading comment
* Insert hard line breaks between declarators if any declarator has an initializer.

## Test Plan

Verified the changes to the existing tests.

File Based Average Prettier Similarity: 77.98% -> 78.00%
Line Based Average Prettier Similarity: 73.23% -> 73.26%

### CODE REVIEW

File: crates/rome_js_formatter/tests/specs/prettier/typescript/as/assignment2.ts.snap [hunk 1]
```diff
@@ -61,8 +61,8 @@ const originalPrototype = originalConstructor.prototype as TComponent & Injectio
 -const originalPrototype = originalConstructor.prototype as TComponent &
 -    InjectionTarget,
 +const originalPrototype = originalConstructor.prototype as
-+  & TComponent
-+  & InjectionTarget,
++    & TComponent
```
Comment (MichaReiser) [inline crates/rome_js_formatter/tests/specs/prettier/typescript/as/assignment2.ts.snap:64]: This is an issue with our formatting of intersection and union types.

**Comment by IWANABETHATGUY** [comment]
Do we still need the prettier compatibility? since the test workflow has changed.

File: crates/rome_js_formatter/src/js/lists/variable_declarator_list.rs [hunk 2]
```diff
@@ -12,50 +11,60 @@ impl FormatRule<JsVariableDeclaratorList> for FormatJsVariableDeclaratorList {
...
 
     fn fmt(&self, node: &JsVariableDeclaratorList, f: &mut JsFormatter) -> FormatResult<()> {
-        let last_index = node.len().saturating_sub(1);
+        let format_inner = format_with(|f| {
+            let length = node.len();
 
-        let mut declarators = node.elements().enumerate().map(|(index, element)| {
-            format_with(move |f| {
-                write!(f, [&element.node().format()])?;
+            let is_parent_for_loop = node.syntax().grand_parent().map_or(false, |parent| {
```
Comment (ematipico) [inline crates/rome_js_formatter/src/js/lists/variable_declarator_list.rs:17]: nit: `|grand_parent|`

File: crates/rome_js_formatter/tests/specs/prettier/js/comments/variable_declarator.js.snap [hunk 3]
```diff
@@ -119,8 +119,10 @@ const foo3 = 123
  
 -let // Comment
 -  foo2 = "val",
--  bar = "val";
-+let foo2 = "val", bar = "val"; // Comment
++let foo2 =
++    // Comment
++    "val",
```
Comment (ematipico) [inline crates/rome_js_formatter/tests/specs/prettier/js/comments/variable_declarator.js.snap:124]: Is the indentation expected?

**Review by ematipico — APPROVED** [review]

**Comment by ematipico** [comment]
Could you please review the title of the PR before merging? It's truncated

File: crates/rome_js_formatter/tests/specs/prettier/js/comments/variable_declarator.js.snap (hunk 3 above)
Comment (MichaReiser) [reply crates/rome_js_formatter/tests/specs/prettier/js/comments/variable_declarator.js.snap:124]: Yes, Prettier does the same [playground](https://prettier.io/playground/#N4Igxg9gdgLgprEAuEAbOMAEB6bmDCEAtkQjADpSaYBmEEATJgLyYDkAbgIapsA0lagCMuAJxbtuvANwg+ICAAcYAS2gBnZKDGiIAdwAKYhJpQ89XAJ6b5Q0VzABrDAGUupADIqocZDR7qcLb2Tq6KDt4A5sgwogCuQSBwREJwACZp6R5cUJFxXJFwAGIQokRcMKq5yCBccTAQciAAFjBEqADqzSrw6uFgcC4mPSocPZY1YOo2IN6BojAG9pHlfgGJAFbqAB4uUegAinEQ8GuogfLhovM1IqmoTYqi3jAdKmkwzcgAHAAMl7pAh17Ioak84PMOL55ABHY7wJZKUy1dQAWh86XSTVEcDhKhxSwKqyQ-nOiUCRBUMXi5P2cCOJ18JPW8hgXCEbw+XyQDFZ9hUqCihBIXBqPAe8jigQAKuzTKSLiAOAkAJJQTKwFxgZ7KACC6pcMEs6DOgQAvmagA)

**Pull request merged**